docs/src/developing/programming-model/transactions.md

```diff
@@ -171,6 +171,33 @@ that this method only supports fixed sized types. Token utilizes the
 trait to encode/decode instruction data for both token instructions as well as
 token account states.
 
+### Multiple instructions in a single transaction
+
+A transaction can contain any number of arbitrary instructions up to the
```

Comment:
This section could more clearly describe the runtime behavior of account deinitialization and what programs should do to harden themselves.  The last two paragraphs give good info about inter vs intra which is really the crux of what the programs need to be aware of.